### DISCUSSION TITLE
inifnteQuery looping item by item

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#3617] [Q&A]
Hello, i have a use case where we used the useInfinteQuery hook to fetch Data.

We want to implement a slider when u can view the query stored items 1 by one.... where the initial item is the one you clicked on from the list .

Basically with next and prev buttons.

Also, when reaching the last item in the infinite list we want to fetch the nextPage and give the user the ability to keep moving back and forth from the updated list.. 

Sounds simple but we reached a few roadblocks especially because we are trying to pass the entire list to another component with the index of the currently selected item,..

any help ,will be much appreciated

**Comment by TkDodo** [comment]
sounds like you're looking for the paginate example:
- https://react-query.tanstack.com/guides/paginated-queries
- https://react-query.tanstack.com/examples/pagination
